java/com/google/re2j/UnicodeTables.java

```diff
@@ -5285,71 +5285,80 @@
-    {0x017F, 0x0053},
-    {0x01C4, 0x01C5},
-    {0x01C5, 0x01C6},
-    {0x01C6, 0x01C4},
-    {0x01C7, 0x01C8},
-    {0x01C8, 0x01C9},
-    {0x01C9, 0x01C7},
-    {0x01CA, 0x01CB},
-    {0x01CB, 0x01CC},
-    {0x01CC, 0x01CA},
-    {0x01F1, 0x01F2},
-    {0x01F2, 0x01F3},
-    {0x01F3, 0x01F1},
-    {0x0345, 0x0399},
-    {0x0392, 0x03B2},
-    {0x0395, 0x03B5},
-    {0x0398, 0x03B8},
-    {0x0399, 0x03B9},
-    {0x039A, 0x03BA},
-    {0x039C, 0x03BC},
-    {0x03A0, 0x03C0},
-    {0x03A1, 0x03C1},
-    {0x03A3, 0x03C2},
-    {0x03A6, 0x03C6},
-    {0x03A9, 0x03C9},
-    {0x03B2, 0x03D0},
-    {0x03B5, 0x03F5},
-    {0x03B8, 0x03D1},
-    {0x03B9, 0x1FBE},
-    {0x03BA, 0x03F0},
-    {0x03BC, 0x00B5},
-    {0x03C0, 0x03D6},
-    {0x03C1, 0x03F1},
-    {0x03C2, 0x03C3},
-    {0x03C3, 0x03A3},
-    {0x03C6, 0x03D5},
-    {0x03C9, 0x2126},
-    {0x03D0, 0x0392},
-    {0x03D1, 0x03F4},
-    {0x03D5, 0x03A6},
-    {0x03D6, 0x03A0},
-    {0x03F0, 0x039A},
-    {0x03F1, 0x03A1},
-    {0x03F4, 0x0398},
-    {0x03F5, 0x0395},
-    {0x1E60, 0x1E61},
-    {0x1E61, 0x1E9B},
-    {0x1E9B, 0x1E60},
-    {0x1E9E, 0x00DF},
-    {0x1FBE, 0x0345},
-    {0x2126, 0x03A9},
-    {0x212A, 0x004B},
-    {0x212B, 0x00C5},
-  };
+  static final char[] CASE_ORBIT = makeCaseOrbit();
+
+  private static char[] makeCaseOrbit() {
+    final char[][] tmp = {
+      {0x004B, 0x006B},
+      {0x0053, 0x0073},
+      {0x006B, 0x212A},
+      {0x0073, 0x017F},
+      {0x00B5, 0x039C},
+      {0x00C5, 0x00E5},
+      {0x00DF, 0x1E9E},
+      {0x00E5, 0x212B},
+      {0x0130, 0x0130},
+      {0x0131, 0x0131},
+      {0x017F, 0x0053},
+      {0x01C4, 0x01C5},
+      {0x01C5, 0x01C6},
+      {0x01C6, 0x01C4},
+      {0x01C7, 0x01C8},
+      {0x01C8, 0x01C9},
+      {0x01C9, 0x01C7},
+      {0x01CA, 0x01CB},
+      {0x01CB, 0x01CC},
+      {0x01CC, 0x01CA},
+      {0x01F1, 0x01F2},
+      {0x01F2, 0x01F3},
+      {0x01F3, 0x01F1},
+      {0x0345, 0x0399},
+      {0x0392, 0x03B2},
+      {0x0395, 0x03B5},
+      {0x0398, 0x03B8},
+      {0x0399, 0x03B9},
+      {0x039A, 0x03BA},
+      {0x039C, 0x03BC},
+      {0x03A0, 0x03C0},
+      {0x03A1, 0x03C1},
+      {0x03A3, 0x03C2},
+      {0x03A6, 0x03C6},
+      {0x03A9, 0x03C9},
+      {0x03B2, 0x03D0},
+      {0x03B5, 0x03F5},
+      {0x03B8, 0x03D1},
+      {0x03B9, 0x1FBE},
+      {0x03BA, 0x03F0},
+      {0x03BC, 0x00B5},
+      {0x03C0, 0x03D6},
+      {0x03C1, 0x03F1},
+      {0x03C2, 0x03C3},
+      {0x03C3, 0x03A3},
+      {0x03C6, 0x03D5},
+      {0x03C9, 0x2126},
+      {0x03D0, 0x0392},
+      {0x03D1, 0x03F4},
+      {0x03D5, 0x03A6},
+      {0x03D6, 0x03A0},
+      {0x03F0, 0x039A},
+      {0x03F1, 0x03A1},
+      {0x03F4, 0x0398},
+      {0x03F5, 0x0395},
+      {0x1E60, 0x1E61},
+      {0x1E61, 0x1E9B},
+      {0x1E9B, 0x1E60},
+      {0x1E9E, 0x00DF},
+      {0x1FBE, 0x0345},
+      {0x2126, 0x03A9},
+      {0x212A, 0x004B},
+      {0x212B, 0x00C5},
+    };
+    final char[] result = new char[tmp[tmp.length - 1][0] + 1];
```

Comment:
A comment is necessary here.

// Precompute the case folding mapping to avoid binary search at run time.
// The 'result' array maps each cased char to the next char in its orbit.
// The orbit is a cycle such as  k -> K -> K [Kelvin] -> k.